client/rpc/src/state/state_full.rs

```diff
@@ -358,6 +361,22 @@ where
 		))
 	}
 
+	fn storages(
+		&self,
+		block: Option<Block::Hash>,
+		keys: Vec<StorageKey>,
+	) -> FutureResult<Vec<Option<StorageData>>> {
+		let block = match self.block_or_best(block) {
+			Ok(b) => b,
+			Err(e) => return Box::new(result(Err(client_err(e)))),
+		};
+		let client = self.client.clone();
+		Box::new(join_all(
+			keys.into_iter()
+				.map(move |key| client.storage(&BlockId::Hash(block), &key).map_err(client_err)),
+		))
```

Comment:
```suggestion
		Box::new(keys.into_iter()
				.map(move |key| client.storage(&BlockId::Hash(block), &key).map_err(client_err))
				.collect::<Result<_, _>>()
		)
```

There are no futures involved.